docs/rules/lines-between-class-members.md

```diff
@@ -0,0 +1,152 @@
+# enforce lines between class members (lines-between-class-members)
+
+This rule improves readability by enforcing lines between class members.
+
+## Rule Details
+
+Examples of **incorrect** code for this rule:
+
+```js
+/* eslint lines-between-class-members: ["error", "always"]*/
+class MyClass {
+  foo() {
+    //...
+  }
+  bar() {
+    //...
+  }
+}
+```
+
+Examples of **correct** code for this rule:
+
+```js
+/* eslint lines-between-class-members: ["error", "always"]*/
+class MyClass {
+  foo() {
+    //...
+  }
+
+  bar() {
+    //...
+  }
+}
+```
+
+### Options
+
+This rule has one option, which can be a string option or an object option.
+
+String option:
+
+* `"always"`(default) requires line breaks between methods
+* `"never"` disallows line breaks between methods
+
+Object option:
+
+* `"multiline": "always"` requires line breaks if the method is multiline
+* `"multiline": "never"` disallow line breaks if the method is multiline
+* `"singleline": "always"` requires line breaks if the method is singleline
+* `"singleline": "never"` disallow line breaks if the method is singleline
```

Comment:
Nitpick: Maybe it would be better to use camelCase for the option names: `multiLine` and `singleLine`

Reply:
I think either is fine -- some rule has the same option. https://github.com/eslint/eslint/commit/7ebd9d6f34b26cd16b478adc8ac7fb2d2c515b11#diff-a99a24653e256e6f5477709c8257845fR18

Reply:
Agreed! Camel case would keep it consistent with our other rules.

Reply:
I'm a little worried that we had some rule option `multiline`:
![image](https://user-images.githubusercontent.com/13050025/30223606-2fe2eb12-94fe-11e7-9a59-7466494d9818.png)

Reply:
It does look like `multiline` is the precedent we've set. Perhaps we should continue using it for consistency. We can make in an issue to update them all in the future in a backwards compatible way and then update the docs to use camel casing.